x-pack/plugins/monitoring/public/directives/apm/instance/index.js

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import moment from 'moment';
+import { render } from 'react-dom';
+import { uiModules } from 'ui/modules';
+import { ApmServerInstance } from 'plugins/monitoring/components/apm/instance';
+import { timefilter } from 'ui/timefilter';
+
+const uiModule = uiModules.get('monitoring/directives', []);
+uiModule.directive('monitoringApmServerInstance', () => {
+  return {
+    restrict: 'E',
```

Comment:
What does `restrict: 'E'` mean?

Reply:
This is old angular knowledge that we can hopefully forget soon, but it means the directive can only be used as an element, like `<monitoring-apm-server-instance>` since angular allows directives to be used by class name or attribute too.

Reply:
![image](https://user-images.githubusercontent.com/51061/44281015-4e7db000-a20b-11e8-9abc-7872d1426576.png)